src/Addons/CategoryFetcher.php

```diff
@@ -0,0 +1,240 @@
+        }
+
+        return null !== $category ? $category : $this->defaultData;
+    }
+
+    /**
+     * @param array $categories
+     * @param int $searchedCategoryId
+     *
+     * @return array|null
+     */
+    private function searchCategory(array $categories, $searchedCategoryId)
+    {
+        foreach ($categories as $category) {
+            if (!empty($category['id_category']) && $searchedCategoryId == $category['id_category']) {
+                return $category;
+            }
+
+            if (!empty($category['categories'])) {
+                $subCategory = $this->searchCategory($category['categories'], $searchedCategoryId);
+                if (null !== $subCategory) {
+                    return $subCategory;
+                }
+            }
+        }
+
+        return null;
+    }
+
+    /**
+     * @param array $category
+     *
+     * @return string
+     */
+    private function getDescription(array $category)
+    {
+        $defaultDescription = !empty($this->defaultData['description']) ? $this->defaultData['description'] : '';
+        //Clean link used to fetch description (no need for tracking then)
+        if (empty($category['clean_link'])) {
+            return $defaultDescription;
+        }
+
+        $circuitBreaker = $this->factory->create([
+            'closed' => [self::CLOSED_ALLOWED_FAILURES, self::PLATFORM_TIMEOUT_SECONDS, 0],
+            'client' => [
+                'method' => 'GET',
+            ],
+        ]);
+
+        $categoryResponse = $circuitBreaker->call($category['clean_link'], function () { return false; });
+        if (empty($categoryResponse)) {
+            return $defaultDescription;
+        }
+
+        $cssSelector = new CssSelectorConverter();
+        $document = new DOMDocument();
+        $document->loadHTML($categoryResponse);
+        $xpath = new DOMXPath($document);
+        $descriptionNode = $xpath->query($cssSelector->toXPath('#category_description'))->item(0);
+        $categoryDescription = '';
+        /** @var DOMNode $childNode */
+        foreach ($descriptionNode->childNodes as $childNode) {
+            $categoryDescription .= $childNode->ownerDocument->saveHTML($childNode);
+        }
+
+        return !empty($categoryDescription) ? $categoryDescription : $defaultDescription;
+    }
+
+    /**
+     * Updates link property with a correctly formatted url with tracking parameters
+     *
+     * @param array $category
+     * @param string $isoCode
+     *
+     * @return array
+     */
+    private function addTracking(array $category, $isoCode)
+    {
+        if (empty($category['link'])) {
+            return $category;
+        }
+
+        $parsedUrl = parse_url($category['link']);
+        if (false === $parsedUrl) {
+            return $category;
+        }
+
+        $parameters = [];
+        if (!empty($parsedUrl['query'])) {
+            parse_str($parsedUrl['query'], $parameters);
```

Comment:
It's better to combine parse_str and parse_url
```suggestion
            parse_str(parse_url($parsedUrl['query'], PHP_URL_QUERY), $parameters);
```

Reply:
is it really useful, since we already have `$parsedUrl = parse_url($category['link']);` why do it twice?

Reply:
oh you're right, my bad!